### ISSUE TITLE
WebDAV sync with Nextcloud overwrites existing Nextcloud session

### ISSUE THREAD

**Issue by Programie** [Tampermonkey/tampermonkey#1054]
I've configured my Nextcloud instance as WebDAV server used to sync my Tampermonkey scripts. In the same browser session, I'm accessing the Nextcloud frontend.

Whenever Tampermonkey syncs with the WebDAV server, the Nextcloud session is overwritten resulting in some weird effects in Nextcloud (i.e. permission denied in some areas of Nextcloud as Tampermonkey is using an application password due to 2FA enabled in my Nextcloud account).

Is there some way to prevent Tampermonkey overwriting the browser session?

## Expected Behavior

Tampermonkey should sync with the WebDAV server without destroying existing sessions.

## Actual Behavior

The WebDAV client of Tampermonkey overwrites existing sessions.

## Specifications

- Chrome 86.0.4240.80
- TM: 4.11
- OS: macOS 10.15.7

**Comment by iav** [comment]
I set tampermonkey to mynextcloud server? it show "sync" then no error. But no any changes on webdav server, just nothing.

**Comment by Programie** [comment]
After three years, I tried to sync with Nextcloud via WebDAV once again. But I still have the same issues as described three years ago. Once I sync with an app password, I'm also logged in with that app password in Nextcloud resulting in some features being disabled (i.e. listing active sessions).

Another extension I'm using (floccus) also synchronizes with my Nextcloud instance but does not overwrite my existing session.

@iav I was able to reproduce a similar issue. In my case, the credentials are incorrect. For some reason, Tampermonkey does not complain about that and just shows "Sync finished". I can see the failed login in the nextcloud.log file. Maybe that's the same issue you are facing?

**Comment by iav** [comment]
No, not my case.
All I see on server log
```
app_1  | 192.168.1.xx - iii [10/Jan/2024:19:27:58 +0000] "OPTIONS /remote.php/dav/files/iii/ HTTP/1.0" 401 2099 "-" "Tampermonkey"
app_1  | 192.168.1.xx - iii [10/Jan/2024:19:28:00 +0000] "PROPFIND /remote.php/dav/files/iii/Tampermonkey/sync/ HTTP/1.0" 401 2093 "-" "Tampermonkey"
app_1  | 192.168.1.xx - iii [10/Jan/2024:19:28:01 +0000] "MKCOL /remote.php/dav/files/iii/Tampermonkey/ HTTP/1.0" 401 2099 "-" "Tampermonkey"
```

**Comment by derjanb** [comment]
> Whenever Tampermonkey syncs with the WebDAV server, the Nextcloud session is overwritten
> Tampermonkey does not complain about that and just shows "Sync finished".

Both will be fixed at the next BETA release.

> app_1  | 192.168.1.xx - iii [10/Jan/2024:19:27:58 +0000] "OPTIONS /remote.php/dav/files/iii/ HTTP/1.0" 401 2099 "-" "Tampermonkey"

`401` stands for `Unauthorized`. So the credentials seem to be wrong?

**Comment by iav** [comment]
Credentials was copy/paste from text file, used by other programs and have webdav access. Maybe, credentials corrupted on paste or when used — I can't see what password stored and submitted by addon.
I will do this test again and will post is something changes.

**Comment by iav** [comment]
Next try:
```
app_1  | 192.168.1.xx - iii [11/Jan/2024:16:16:04 +0000] "OPTIONS /remote.php/dav/files/iii/ HTTP/1.0" 200 1983 "-" "Tampermonkey"
app_1  | 192.168.1.xx - iii [11/Jan/2024:16:16:04 +0000] "PROPFIND /remote.php/dav/files/iii/Tampermonkey/sync/ HTTP/1.0" 404 2042 "-" "Tampermonkey"
app_1  | 192.168.1.xx - iii [11/Jan/2024:16:16:05 +0000] "MKCOL /remote.php/dav/files/iii/Tampermonkey/ HTTP/1.0" 405 1956 "-" "Tampermonkey"
```
Try to list backups in tab "utilites", got list of backups
```
app_1  | 192.168.1.xx - iii [11/Jan/2024:16:18:37 +0000] "OPTIONS /remote.php/dav/files/iii/ HTTP/1.0" 200 1961 "-" "Tampermonkey"
app_1  | 192.168.1.xx - iii [11/Jan/2024:16:18:38 +0000] "PROPFIND /remote.php/dav/files/iii/Tampermonkey/ HTTP/1.0" 207 2245 "-" "Tampermonkey"
app_1  | 192.168.1.xx - iii [11/Jan/2024:16:18:38 +0000] "PROPFIND /remote.php/dav/files/iii/Tampermonkey/ HTTP/1.0" 207 2643 "-" "Tampermonkey"
```

**Comment by derjanb** [comment]
405 means not allowed to create the Tampermonkey folder, but it looks like it is there at the second log. Will check.

**Comment by iav** [comment]
There is folder `Tampermonkey`, there are some backups I did before and now. There no `sync` folder inside.

**Comment by derjanb** [comment]
> there are some backups I did before [...]. There no sync folder inside.

I found the issue. Thanks. Will be fixed at the next BETA version as well.

**Comment by iav** [comment]
Yes, seems sync start to work after I create `sync` folder.

**Comment by derjanb** [comment]
Should be fixed at 5.1.6193 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi in review)

Chrome/Edge users, please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions` (after you've enabled 'Developer Mode').
Firefox users please install the [BETA version](https://www.tampermonkey.net/index.php?browser=firefox) or check for BETA version updates at `about:addons` 

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.
